Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -7585,20 +7585,71 @@ RefPtr<CSSStyleValue> Parser::parse_grid_shorthand_value(TokenStream<ComponentVa
 RefPtr<CSSStyleValue> Parser::parse_grid_template_areas_value(TokenStream<ComponentValue>& tokens)
 {
     // none | <string>+
-    Vector<Vector<String>> grid_area_rows;
-
     if (auto none = parse_all_as_single_keyword_value(tokens, Keyword::None))
-        return GridTemplateAreaStyleValue::create(move(grid_area_rows));
+        return GridTemplateAreaStyleValue::create({});
+
+    auto is_whitespace = [](u32 code_point) {
+        return code_point == '\n' || code_point == '\t' || code_point == ' ';
+    };
+
+    auto is_full_stop = [](u32 code_point) {
+        return code_point == '.';
+    };
+
+    auto is_ident_code_point = [](u32 code_point) {
+        return is_ascii_alphanumeric(code_point)
+            || code_point == '_'
+            || code_point == '-'
+            || code_point >= 0x0080;
+    };
```

Comment:
As these are duplicate definitions of those found in `Tokenizer.cpp`, maybe those could be moved out somewhere accessible instead of duplicating the logic here.

Reply:
Good idea, `Libraries/LibWeb/Infra/CharacterTypes.h` and `AK/CharacterTypes.h` seem like viable homes for those definitions, not sure which I should prefer.

Reply:
I've moved `is_whitespace`, `is_ident_start_code_point` and `is_ident_code_point` to a new `Libraries/LibWeb/CSS/CharacterTypes.h` header. The other definitions can be moved later, but let me know if you want me to move any more in this PR.